Sources/Target/Windows/Process.cpp

```diff
@@ -331,6 +331,18 @@ ErrorCode Process::readMemory(Address const &address, void *data, size_t length,
 
 ErrorCode Process::writeMemory(Address const &address, void const *data,
                                size_t length, size_t *nwritten) {
+  // We try to commit the necessary pages in case they have been uncommitted.
+  // Otherwise writes may fail.
+  LPVOID allocError = VirtualAllocEx(
+      _handle,
+      reinterpret_cast<LPVOID>(address.value()),
+      length,
+      MEM_COMMIT,
+      PAGE_EXECUTE_READWRITE);
+  if (allocError == NULL) {
+    return Platform::TranslateError();
+  }
+
```

Comment:
This changes the "behaviour" of the memory (from the debuggee' perspective). Aside from turning an exception into a valid read/write (__chk_stack?), you're changing the protection of the memory.

This also will allocate memory if it was never allocated, which I'd imagine would hide bugs.

In what cases are we trying to write to virtual addresses without committed memory behind them?